### ISSUE TITLE
API for netwo

### ISSUE THREAD

**Issue by MrBlenny** [KeitIG/museeks#324]

**Comment by MrBlenny** [comment]
Another change that would be to add a redux-store to the main-thread using something like [electron-redux](https://github.com/hardchor/electron-redux). This makes sure the redux state is synced between the main-thread and renderers.

This would be particularly because the API (running in the main-thread) does not currently have access to the redux store so it is relatively complicated to return data in the redux store (as we must run a function in the renderer in order to do so).

Not sure what you think of this...

**Comment by YurySolovyov** [comment]
Depends on how much data you need to be shared, ATM it is not much if any

**Comment by KeitIG** [comment]
> Remote playback

Hell yes

> Shared library

I'll let this one up to you :smile: 

===

About `electron-redux`, if it can solve the problem nicely, why not ?

**Comment by YurySolovyov** [comment]
What's the plan on this remote service discovery?
I guess that would need a UI to copy link or whatever...

**Comment by MrBlenny** [comment]
Yep, I think network discovery will be on the SettingsLibrary page but I'll give it some more thought. 

It will automatically discover other instances of Museeks running on the same network by pinging all IPs in its range X.X.X.1-255:SomePort
Might be a good idea to add a pin/password for auth but I'll leave that for later.

**Comment by YurySolovyov** [comment]
> by pinging all IPs in its range X.X.X.1-255:SomePort

That sounds expensive

**Comment by KeitIG** [comment]
I don't really see other ways to be honest

**Comment by YurySolovyov** [comment]
There are whisper protocols there, but they would need investigations...
Though remote access does not have to be Skynet scale right away, we can start just by the ability to copy and paste a link...

**Comment by KeitIG** [comment]
https://github.com/eviltik/evilscan

**Comment by DrEVILish** [comment]
> Yep, I think network discovery will be on the SettingsLibrary page but I'll give it some more thought.
> It will automatically discover other instances of Museeks running on the same network by pinging all IPs in its range X.X.X.1-255:SomePort
> Might be a good idea to add a pin/password for auth but I'll leave that for later.

Look into the Broadcast address which is part of the IP spec. I think this is how it works, I might be wrong.

We could have as Museeks opens it broadcasts a "HELLO" packet with some relevant information about itself on the Broadcast address. If it's the first one no one will hear it. (Tree Fall Woods)
If there is another app open on the network, it will receive the "HELLO" and send a "REPLY" with some information to Sync the two or more servers together with information about each other.
